uber/models.py

```diff
@@ -439,14 +439,18 @@ def order(self, attrs):
                 order.append(col.desc() if attr.startswith('-') else col)
             return self.order_by(*order)
 
-        def icontains(self, attr=None, val=None, **filters):
-            query = self
+        def icontains_predicate(self, attr=None, val=None, **filters):
```

Comment:
I've read through this a few times, and I can't tell what the difference is between the original ``.icontains`` and ``.icontains_predicate``.  The original version basically created the equivalent of e.g.

```
.filter(Attendee.first_name.icontains(s)).filter(Attendee.last_name.icontains(s))
```

whereas this version creates the equivalent of

```
.filter(and_([Attendee.first_name.icontains(s), Attendee.last_name.icontains(s)]))
```

but those two are equivalent in every way I know about.  What's the reason for the change?

Regardless of the answer, please leave a docstring comment in the code explaining why the function exists.  I always feel weird asking people to do that, since I've so often neglected to do it myself, but we're trying to get better about it, and this is the kind of "probably not intuitive at first glance" method that really needs one.